Yosemite/Yosemite/Stores/SitePluginStore.swift

```diff
@@ -47,12 +53,68 @@ private extension SitePluginStore {
             }
         }
     }
+
+    func installSitePlugin(siteID: Int64, slug: String, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        remote.installPlugin(for: siteID, slug: slug) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let plugin):
+                self.upsertSitePluginsInBackground(siteID: siteID, readonlyPlugins: [plugin], completionHandler: onCompletion)
+            case .failure(let error):
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    func activateSitePlugin(siteID: Int64, pluginName: String, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        remote.activatePlugin(for: siteID, pluginName: pluginName) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let plugin):
+                guard plugin.status == .active else {
+                    onCompletion(.failure(SitePluginError.activationFailed))
+                    return
+                }
+                self.upsertSitePluginsInBackground(siteID: siteID, readonlyPlugins: [plugin], completionHandler: onCompletion)
+            case .failure(let error):
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    func getPluginDetails(siteID: Int64, pluginName: String, onCompletion: @escaping (Result<SitePlugin, Error>) -> Void) {
+        remote.getPluginDetails(for: siteID, pluginName: pluginName) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let plugin):
+                self.upsertSitePluginInBackground(siteID: siteID, readonlyPlugin: plugin, completionHandler: onCompletion)
+            case .failure(let error):
+                onCompletion(.failure(error))
+            }
+        }
+    }
 }
 
 // MARK: - Storage
 //
 private extension SitePluginStore {
 
+    /// Updates or inserts Readonly the `SitePlugin` entity in background.
+    /// Triggers `completionHandler` on main thread.
+    ///
+    func upsertSitePluginInBackground(siteID: Int64, readonlyPlugin: SitePlugin, completionHandler: @escaping (Result<SitePlugin, Error>) -> Void) {
```

Comment:
do you think it makes sense to call `upsertSitePluginsInBackground` by passing `[readonlyPlugin]` to the  `readonlyPlugins` parameter and call `completionHandler(.success(readonlyPlugin))` on completion?

Reply:
Oh right! I didn't think about this before 😅 Updated in 542d8837cef7369202adc58be8e06cc636dba662.